Extension/src/Utility/System/info.ts

```diff
@@ -0,0 +1,145 @@
+    const parent = Object.getPrototypeOf(typeof instance === 'function' ? instance : instance.constructor);
+    return parent.name ? parent : undefined;
+}
+
+export function classOf(instance: AribtraryObject | Constructor): Constructor | undefined {
+    return instance ? typeof instance === 'function' ? // is it a javascript function of some kind?
+        is.asyncConstructor(instance) ? classOf(instance.class) :
+            is.Constructor(instance) ? instance as Constructor // is it really a constructor?
+                : undefined  // no, it's a js function, but not a constructor
+        : instance.constructor as Constructor : // it's an object, so get the constructor from the object
+        undefined;
+}
+
+/** returns true if the instance is an anonymous object (as opposed to constructed via a class) */
+export function isAnonymousObject(instance: any): boolean {
+    return instance.constructor.name === 'Object';
+}
+
+interface FunctionInfo {
+    /** was this declared with quotes or other non-word characters in the function name */
+    hasNonWordCharacters: boolean;
+
+    /** is the function async */
+    isAsync: boolean;
+
+    /** a bound callable function for the member (saves us from having to do it later anyway) */
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    fn: Function;
+}
+
+interface Members {
+    methods: Map<string, FunctionInfo>;
+    fields: Map<string, string>;
+    properties: Map<string, string>;
+}
+
+const builtIns = new Set(Object.keys(Object.getOwnPropertyDescriptors(Object.getPrototypeOf({}))));
+
+export function members(obj: any): Members {
+    const result = {
+        methods: new Map<string, FunctionInfo>(),
+        fields: new Map<string, string>(),
+        properties: new Map<string, string>()
+    };
+
+    if (typeof obj === 'object') {
+        let instance = obj;
+        do {
+            // enumerate all the properties at this level.
+            for (const [memberName, descriptor] of Object.entries(Object.getOwnPropertyDescriptors(instance))) {
+                // check if we're filtering it out because it's built into object, or it's using a leading underscore.
+                if (!(builtIns.has(memberName) || memberName.startsWith('_') || memberName.startsWith('$'))) {
+                    let value: any;
+
+                    // look at the value/type of the member
+                    try {
+                        value = descriptor.value === undefined ? obj[memberName] : descriptor.value;
+                    } catch {
+                        continue;
+                    }
+                    const type = typeof (value);
+
+                    // is it a function
+                    if (type === 'function') {
+                        // it is actually possible to get the 'declared name' of a member (if it was quoted in the source code)
+                        // const declaredName = obj[memberName].toString().replace(/\([\S\s]*/gm,'') || memberName;

```

Comment:
It seems like `// const declaredName` is intentional -- but could you add a few words explaining the current situation, such as "We don't need it now, but it may be useful in the future." or something?